keps/74-mooncake-integration/README.md

```diff
@@ -0,0 +1,388 @@
+| sglang-router [Optional]                      | Orchestrates request routing and load balancing across the system. It directs incoming requests to appropriate sglang-prefill or sglang-decode instances, coordinates handoff between prefill and decode stages, and ensures efficient, consistent access to KVCache data via the Mooncake storage layer.           |
+| sglang-engine [sglang-prefill, sglang-decode] | Integrates an inference engine along with a Mooncake store. The embedded Mooncake store supports multiple modes, including Client-Only mode, which does not allocate local memory for caching and operates solely as a client issuing KVCache Put/Get requests.    |
+
+
+#### Deploy standalone inference service with mooncake
+
+By following the steps below, you can create a SGLang inference service with three Mooncake store replicas, each provisioned with 10Gi of memory, serving as the distributed L3 KVCache storage.
+```yaml
+apiVersion: workloads.x-k8s.io/v1alpha1
+kind: RoleBasedGroup
+metadata:
+  name: sglang-demo
+spec:
+  roles:
+    - name: mooncake-master
+      replicas: 1
+      template:
+        spec:
+          containers:
+            - name: master
+              image: lmsysorg/sglang:v0.5.3.post1
+              env:
+                - name: POD_IP
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: status.podIP
+              command:
+                - sh
+                - -c
+                - "mooncake_master \
+                --rpc_address $(POD_IP) --rpc_port 50051 \
+                --http_metadata_server_host $(POD_IP) --http_metadata_server_port 8080 \
+                --metrics_port 9003"
+              ports:
+                - containerPort: 50051
+                  name: http
+              readinessProbe:
+                initialDelaySeconds: 10
+                periodSeconds: 10
+                tcpSocket:
+                  port: 50051
+    - name: mooncake-store
+      replicas: 3
+      template:
+        spec:
+          containers:
+            - name: store
+              image: lmsysorg/sglang:v0.5.3.post1
+              env:
+                - name: MOONCAKE_MASTER
+                  value: "sglang-demo-mooncake-master-0.sglang-demo-mooncake-master:50051"
+                - name: MOONCAKE_TE_META_DATA_SERVER
+                  value: "http://sglang-demo-mooncake-master-0.sglang-demo-mooncake-master:8080/metadata"
+                - name: MOONCAKE_GLOBAL_SEGMENT_SIZE
+                  value: "10gb"
+              command:
+                - sh
+                - -c
+                - "python -m mooncake.mooncake_store_service"
+              resources:
+                limits:
+                  memory: "10Gi"
+                requests:
+                  memory: "10Gi"
```

Comment:
Can we raise this number? 10G will cause the mooncake store to hit the watermark too soon.